File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
      let left = 0;
    let right = height.length - 1;
    let result = (right - left)*Math.min(height[left],height[right]);
    while (left <= right)
    {
        result = Math.max((right - left)*Math.min(height[left],height[right]),result);
        if (height[right] < height[left]) right--;
        else left++;
    }
    return result;
};

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int result = (right - left) * min(height[left],height[right]);
        while(left <= right)
        {
            result = max(result,(right - left) * min(height[left],height[right]));
            if (height[right] < height[left]) right--;
            else left++;
        }
        return result;
    
    }
};